factory/glideFactoryEntry.py

```diff
@@ -1222,65 +1242,56 @@ def unit_work_v3(
-            for i in sys.exc_info():
-                entry.log.exception("%s" % i)
-        finally:
-            if os.path.exists(tmpnm):
-                os.remove(tmpnm)
-    if os.path.exists(condortoken_file):
-        if not submit_credentials.add_identity_credential("frontend_condortoken", condortoken_file):
+        condortoken = create_credential(string=force_bytes(condortoken_data), path=condortoken_file)
+        entry.log.info(f"frontend_token supplied, writing to {condortoken.path}")
+        condortoken.save_to_file(backup=True)
+        if not submit_credentials.add_identity_credential("frontend_condortoken", condortoken):
             entry.log.warning(
                 "failed to add frontend_condortoken %s to the identity credentials %s"
-                % (condortoken_file, str(submit_credentials.identity_credentials))
+                % (condortoken.path, str(submit_credentials.identity_credentials))
             )
 
     scitoken_passthru = params.get("CONTINUE_IF_NO_PROXY") == "True"
-    scitoken = f"credential_{client_int_name}_{entry.name}.scitoken"
-    scitoken_file = os.path.join(submit_credentials.cred_dir, scitoken)
+    scitoken_name = f"credential_{client_int_name}_{entry.name}.scitoken"
+    scitoken_file = os.path.join(submit_credentials.cred_dir, scitoken_name)
     scitoken_data = decrypted_params.get("frontend_scitoken")
     if scitoken_data:
-        if token_util.token_str_expired(scitoken_data):
+        scitoken = create_credential(string=force_bytes(scitoken_data), path=scitoken_file)
+        if not scitoken.valid():
             entry.log.warning(f"Continuing, but the frontend_scitoken supplied is expired: {scitoken_file}")
-        tmpnm = ""
-        try:
-            entry.log.info(f"frontend_scitoken supplied, writing to {scitoken_file}")
-            (fd, tmpnm) = tempfile.mkstemp(dir=submit_credentials.cred_dir)
-            with os.fdopen(fd, "w", encoding="utf-8") as f:
-                f.write(f"{scitoken_data.strip()}\n")
-            chmod(tmpnm, 0o600)
-            util.file_tmp2final(scitoken_file, tmpnm)
-        except Exception as err:
-            entry.log.exception(f"failed to create scitoken: {err}")
-        finally:
-            if os.path.exists(tmpnm):
-                os.remove(tmpnm)
-
-    if os.path.exists(scitoken_file):
+        entry.log.info(f"frontend_scitoken supplied, writing to {scitoken_file}")
+        scitoken.save_to_file(backup=True)
         # TODO: why identity_credential and not submit_credential? Consider moving when refactoring
-        if not submit_credentials.add_identity_credential("frontend_scitoken", scitoken_file):
+        if not submit_credentials.add_identity_credential("frontend_scitoken", scitoken):
             entry.log.warning(
                 "failed to add frontend_scitoken %s to identity credentials %s"
                 % (scitoken_file, str(submit_credentials.identity_credentials))
```

Comment:
maybe use scitoken.path() (I know the value is the same ... to be consistent)

Reply:
Done.